test/funkyPlugin.test.js

```diff
@@ -0,0 +1,330 @@
+'use strict'
+
+const fastify = require('fastify')
+const { either, task, taskEither } = require('fp-ts')
+const { test } = require('tap')
+const {
+  initAppGet,
+  assertResponseTypeAndBody,
+  assertCorrectResponse,
+  assertCorrectResponseBody,
+  assertErrorResponse
+} = require('./helpers')
+const { fastifyFunky } = require('../index')
+
+const DUMMY_USER = {
+  user: {
+    id: 1
+  }
+}
+
+test('Promise: Correctly handles top-level promise', async (t) => {
+  t.plan(2)
+
+  const route = (_req, _reply) => {
+    return Promise.resolve(either.right(DUMMY_USER))
+  }
+
+  const app = await initAppGet(t, route).ready()
+
+  t.teardown(() => {
+    app.close()
+  })
+
+  await assertCorrectResponse(t, app)
```

Comment:
not a fan about this helper, but it was already there 🙌🏽

Reply:
I just wanted to keep as much structure as possible the same to not make it impossible to review it without having to understand the whole plugin